integration-tests/helpers/fake-agent.js

```diff
@@ -32,7 +42,14 @@ module.exports = class FakeAgent extends EventEmitter {
   }
 
   stop () {
+    if (!this.server) return Promise.resolve()
+
     return new Promise((resolve) => {
+      if (!this.server?.listening) return resolve()
+      for (const socket of this._sockets) {
+        socket.destroy()
+      }
+      this._sockets.clear()
       this.server.on('close', resolve)
       this.server.close()
     })
```

Comment:
We could make it very readable using `once` from the built-in events module.

```suggestion
  async stop () {
    if (!this.server || !this.server?.listening) return

     for (const socket of this._sockets) {
      socket.destroy()
    }
    this._sockets.clear()
    this.server.close()
    return once(this.server, 'close')
```

Reply:
Addressed in follow up PR: https://github.com/DataDog/dd-trace-js/pull/5854